### PR TITLE
ODF template update

### PR DESCRIPTION
**Pull request by gayathrimenath-ibm**

### CODE REVIEW

**Review by akgunjal — APPROVED** [review]
lgtm

File: config-templates/redhat/odf-local/4.13/custom-parameters.json [hunk 1]
```diff
@@ -196,6 +196,18 @@
     "type": "boolean",
     "required": "true"
   },
+  {
+    "description": "Set to true if you want to disable Noobaa public load balancer",
+    "displayname": "Disable Noobaa LB",
+    "name": "disable-noobaa-LB",
```
Comment (mssachan) [inline config-templates/redhat/odf-local/4.13/custom-parameters.json:202]: where are we using this parameter in deployment.yaml?

File: config-templates/redhat/odf-local/changelog.json [hunk 2]
```diff
@@ -3,43 +3,43 @@
         "4.13": {
             "status": "supported",
             "latest-revision": {
+                "revision": "6",
+                "release-date": "19 January 2023",
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json]: update release date as `25 January 2024`

File: config-templates/redhat/odf-local/changelog.json [hunk 3]
```diff
@@ -3,43 +3,43 @@
...
     {
         "4.12": {
             "status": "supported",
             "latest-revision": {
-                "revision": "8",
-                "release-date": "27 November 2023",
-                "cves": "CVE-2007-4559, CVE-2023-4641, CVE-2023-22745",
+                "revision": "9",
+                "release-date": "19 January 2023",
+                "cves": "",
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json]: There are CVEs reported for older image of 4.12.  Can you please update all CVEs here?

File: config-templates/redhat/odf-local/changelog.json [hunk 4]
```diff
@@ -3,43 +3,43 @@
         "4.13": {
             "status": "supported",
             "latest-revision": {
+                "revision": "6",
+                "release-date": "19 January 2023",
+                "cves": "",
+                "golang": "1.20.11",
+                "ubi": "8.9-1029",
+                "new-features": ["Enable/Disable Noobaa public LB"],
+                "fixes": ["fixed uninstallation issues"]
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json]: can you run `json pretty` for this file?
Or manually replace
```
"new-features": ["Enable/Disable Noobaa public LB"],
"fixes": ["fixed uninstallation issues"]
```
---->
```
"new-features": [
     "Enable/Disable Noobaa public LB"
     ],
 "fixes": [
      "fixed uninstallation issues"
   ]
```

File: config-templates/redhat/odf-local/changelog.json [hunk 5]
```diff
@@ -3,43 +3,43 @@
         "4.13": {
             "status": "supported",
             "latest-revision": {
+                "revision": "6",
+                "release-date": "19 January 2023",
+                "cves": "",
+                "golang": "1.20.11",
+                "ubi": "8.9-1029",
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json:10]: new ubi image is available. Do you want to upgrade to new UBI image and rebuild the image?

File: config-templates/redhat/odf-local/changelog.json [hunk 6]
```diff
@@ -3,43 +3,43 @@
...
-                "release-date": "27 November 2023",
-                "cves": "CVE-2007-4559, CVE-2023-4641, CVE-2023-22745",
+                "revision": "9",
+                "release-date": "19 January 2023",
+                "cves": "",
                 "golang": "1.20.11",
                 "ubi": "8.9-1029",
                 "new-features": [],
-                "fixes": []
+                "fixes": ["fixed uninstallation issues"]
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json]: same here

File: config-templates/redhat/odf-remote/4.11/storage-class.yaml [hunk 7]
```diff
@@ -35,7 +35,7 @@ items:
       labels:
         app.kubernetes.io/name: "redhat-odf-remote"
         app.kubernetes.io/version: "v4.11.0"
-        app.kubernetes.io/revision: "12"
+        app.kubernetes.io/revision: "13"
```
Comment (mssachan) [inline config-templates/redhat/odf-remote/4.11/storage-class.yaml:38]: update revision at https://github.com/IBM/ibm-satellite-storage/pull/386/files#diff-b380445b3fc0d6003303e79b115c30f428f697d6192b0ee707484ad6ad18ae72R11

File: config-templates/redhat/odf-remote/changelog.json [hunk 8]
```diff
@@ -70,25 +70,7 @@
...
-            "previous-revision": {
-                "revision": "22",
-                "release-date": "30 October 2023",
-                "cves": "CVE-2023-44487",
-                "golang": "1.20.7",
-                "ubi": "8.8-1072.1697626218",
-                "new-features": [],
-                "fixes": []
-            }
+            "status": "unsupported"
```
Comment (mssachan) [inline config-templates/redhat/odf-remote/changelog.json:81]: please update changelog.json for 4.11, 4.12 & 4.13 versions as well.

**Review by mssachan — CHANGES_REQUESTED** [review]
Also get approval from doc team for new param description.

**Comment by gayathrimenath-ibm** [comment]
> Also get approval from doc team for new param description.

https://github.ibm.com/alchemy-containers/documentation/issues/12227 Doc issue created

File: config-templates/redhat/odf-local/changelog.json (hunk 3 above)
Comment (gayathrimenath-ibm) [reply config-templates/redhat/odf-local/changelog.json]: There is no CVE reported for the new UBI image published. As the new UBI is available, I moved to that image

File: config-templates/redhat/odf-local/changelog.json (hunk 5 above)
Comment (gayathrimenath-ibm) [reply config-templates/redhat/odf-local/changelog.json:10]: will do that in the next push along with VPC and classic update

File: config-templates/redhat/odf-local/changelog.json [hunk 9]
```diff
@@ -3,43 +3,49 @@
         "4.13": {
             "status": "supported",
             "latest-revision": {
+                "revision": "6",
+                "release-date": "25 January 2023",
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json]: `25 January 2023` ---> `25 January 2024`

File: config-templates/redhat/odf-local/changelog.json [hunk 10]
```diff
@@ -3,43 +3,49 @@
...
+                "cves": "",
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json]: `"cves": "",` can be empty only if this PR contains parameter changes and no security fix. Please re-verify and update accordingly.

File: config-templates/redhat/odf-local/changelog.json [hunk 11]
```diff
@@ -3,43 +3,49 @@
...
     },
     {
         "4.12": {
             "status": "supported",
             "latest-revision": {
-                "revision": "8",
-                "release-date": "27 November 2023",
-                "cves": "CVE-2007-4559, CVE-2023-4641, CVE-2023-22745",
+                "revision": "9",
+                "release-date": "25 January 2023",
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json]: same here

File: config-templates/redhat/odf-local/changelog.json (hunk 3 above)
Comment (mssachan) [reply config-templates/redhat/odf-local/changelog.json]: Here we have to mention what all CVEs are fixed after moving to new image.
Its not about which CVEs are present in new UBI image.

File: config-templates/redhat/odf-local/changelog.json [hunk 12]
```diff
@@ -49,48 +55,32 @@
         "4.11": {
             "status": "supported",
             "latest-revision": {
-                "revision": "12",
-                "release-date": "27 November 2023",
-                "cves": "CVE-2007-4559, CVE-2023-4641, CVE-2023-22745",
+                "revision": "13",
+                "release-date": "25 January 2023",
```
Comment (mssachan) [inline config-templates/redhat/odf-local/changelog.json]: same here

File: config-templates/redhat/odf-remote/changelog.json [hunk 13]
```diff
@@ -3,43 +3,49 @@
         "4.13": {
             "status": "supported",
             "latest-revision": {
+                "revision": "6",
+                "release-date": "25 January 2023",
```
Comment (mssachan) [inline config-templates/redhat/odf-remote/changelog.json]: same here

**Review by mssachan — CHANGES_REQUESTED** [review]
Added minor comments.

**Comment by gayathrimenath-ibm** [comment]
CVEs were present for previous image, listing the details below. updated the code as well
```
urineedisatyabhargavi@Surineedis-MacBook-Pro-2 alchemy-containers % ibmcloud cr va icr.io/obs/ibm-ocs-operator:v4.11.18
Checking security issues for 'icr.io/obs/ibm-ocs-operator:v4.11.18'...

Image 'icr.io/obs/ibm-ocs-operator:v4.11.18' was last scanned on Mon Jan 22 14:59:23 UTC 2024
The scan results show that 6 ISSUES were found for the image.

Vulnerable Packages Found
=========================

Vulnerability ID   Policy Status   Affected Packages               How to Resolve
CVE-2023-39615     Active          libxml2 and libxml2             Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-5981      Active          gnutls and gnutls               Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-7104      Active          sqlite-libs and sqlite-libs     Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-3446      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-3817      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-5678      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.

To see the details about the fixes for these packages, run the command again with the '--extended' flag.

OK
surineedisatyabhargavi@Surineedis-MacBook-Pro-2 alchemy-containers % ibmcloud cr va icr.io/obs/ibm-ocs-operator:v4.11.19
Checking security issues for 'icr.io/obs/ibm-ocs-operator:v4.11.19'...

Image 'icr.io/obs/ibm-ocs-operator:v4.11.19' was last scanned on Tue Jan 23 04:49:02 UTC 2024
The scan results show that NO ISSUES were found for the image.

OK
surineedisatyabhargavi@Surineedis-MacBook-Pro-2 alchemy-containers % ibmcloud cr va icr.io/obs/ibm-ocs-operator:v4.12.14
Checking security issues for 'icr.io/obs/ibm-ocs-operator:v4.12.14'...

Image 'icr.io/obs/ibm-ocs-operator:v4.12.14' was last scanned on Mon Jan 22 20:59:22 UTC 2024
The scan results show that 6 ISSUES were found for the image.

Vulnerable Packages Found
=========================

Vulnerability ID   Policy Status   Affected Packages               How to Resolve
CVE-2023-39615     Active          libxml2 and libxml2             Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-7104      Active          sqlite-libs and sqlite-libs     Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-5981      Active          gnutls and gnutls               Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-3446      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-3817      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-5678      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.

To see the details about the fixes for these packages, run the command again with the '--extended' flag.

OK
surineedisatyabhargavi@Surineedis-MacBook-Pro-2 alchemy-containers % ibmcloud cr va icr.io/obs/ibm-ocs-operator:v4.12.15
Checking security issues for 'icr.io/obs/ibm-ocs-operator:v4.12.15'...

Image 'icr.io/obs/ibm-ocs-operator:v4.12.15' was last scanned on Tue Jan 23 04:49:02 UTC 2024
The scan results show that NO ISSUES were found for the image.

OK
surineedisatyabhargavi@Surineedis-MacBook-Pro-2 alchemy-containers % 
surineedisatyabhargavi@Surineedis-MacBook-Pro-2 alchemy-containers % ibmcloud cr va icr.io/obs/ibm-ocs-operator:v4.13.7
Checking security issues for 'icr.io/obs/ibm-ocs-operator:v4.13.7'...

Image 'icr.io/obs/ibm-ocs-operator:v4.13.7' was last scanned on Mon Jan 22 09:01:27 UTC 2024
The scan results show that 6 ISSUES were found for the image.

Vulnerable Packages Found
=========================

Vulnerability ID   Policy Status   Affected Packages               How to Resolve
CVE-2023-3446      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-3817      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-5678      Active          openssl-libs and openssl-libs   Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-5981      Active          gnutls and gnutls               Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-7104      Active          sqlite-libs and sqlite-libs     Upgrade 2 packages. Re-run command with --extended to view.
CVE-2023-39615     Active          libxml2 and libxml2             Upgrade 2 packages. Re-run command with --extended to view.

To see the details about the fixes for these packages, run the command again with the '--extended' flag.

OK
surineedisatyabhargavi@Surineedis-MacBook-Pro-2 alchemy-containers % ibmcloud cr va icr.io/obs/ibm-ocs-operator:v4.13.10
Checking security issues for 'icr.io/obs/ibm-ocs-operator:v4.13.10'...

Image 'icr.io/obs/ibm-ocs-operator:v4.13.10' was last scanned on Tue Jan 23 04:49:02 UTC 2024
The scan results show that NO ISSUES were found for the image.

OK

```

**Review by mssachan — APPROVED** [review]
lgtm

**Pull request merged**